dvc/remote/local/__init__.py

```diff
@@ -566,3 +593,78 @@ def unprotect(path_info):
     @staticmethod
     def protect(path_info):
         os.chmod(path_info.path, stat.S_IREAD | stat.S_IRGRP | stat.S_IROTH)
+
+    @classmethod
+    def _append_unpacked_suffix(cls, string):
+        return string + cls.UNPACKED_DIR_SUFFIX
+
+    def _get_unpacked_dir_path_info(self, checksum):
+        unpacked_dir_info = self.checksum_to_path_info(checksum)
+        unpacked_dir_info.path = self._append_unpacked_suffix(
+            unpacked_dir_info.path
+        )
+
+        return unpacked_dir_info
+
+    def _path_info_changed(self, path_info):
+        if self.exists(path_info) and self.state.get(path_info):
+            return False
+        return True
+
+    def _update_unpacked_dir(self, checksum):
+        unpacked_dir_info = self._get_unpacked_dir_path_info(checksum)
+
+        if not self._path_info_changed(unpacked_dir_info):
+            return
+
+        self.remove(unpacked_dir_info)
+
+        try:
+            dir_info = self.get_dir_cache(checksum)
+            self._create_unpacked_dir(checksum, dir_info, unpacked_dir_info)
+        except Exception:
+            logger.warning(
+                "Could not create '{}'".format(unpacked_dir_info.path)
+            )
+
+            self.remove(unpacked_dir_info)
+
+    def _create_unpacked_dir(self, checksum, dir_info, unpacked_dir_info):
```

Comment:
For large directories, it might take a while, so we need to have a progress bar or something here, same as with checkout, but with a more suitable message.

Reply:
https://asciinema.org/a/GcKkT1blIxTzXs8FfwR2kEcvZ

Reply:
https://asciinema.org/a/0Jj6YF8Jxjzn0srxAztf21xgc